addons/sale_expense_margin/models/sale_order_line.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    expense_id = fields.Many2one('hr.expense', string='Expense')
+
+    @api.depends('is_expense')
+    def _compute_purchase_price(self):
+        date_today = fields.Date.context_today(self.env.user)
```

Comment:
```suggestion
        date_today = fields.Date.context_today(self)
```
`context_today` only uses the environment of the given record, giving the user doesn't change anything